### ISSUE TITLE
New `tolerance_linter()` to detect use of `round()` instead of a custom `tolerance` value

### ISSUE THREAD

**Issue by Bisaloo** [r-lib/lintr#1951]
I've noticed the following pattern a couple of times:

```r
all.equal(round(x, 4), 1.2345))
expect_equal(round(x, 4), 1.2345))
```

Examples in CRAN packages:

- https://github.com/search?q=org%3Acran%20all.equal(round(&type=code
- https://github.com/search?q=org%3Acran+expect_equal(round(&type=code

It would be clearer to set a custom `tolerance` value in this case and skip the `round()` call entirely:

```r
all.equal(x, 1.2345, tolerance = t)
expect_equal(x, 1.2345, tolerance = t)
```

I don't think we can give a recommendation for the `tolerance` value in the error message because values may be compared on a relative scale depending on the values in `x`. As a result, the linted expression might not be entirely equivalent to the original one but I believe it makes the desired intent clearer.

One potential valid use case as far as I can tell, even though it's not ideal either, is to check if `x` contains an integer value:

```r
all.equal(round(x), x)
```

**Comment by Bisaloo** [comment]
Now that we have `all_equal_linter()`, should this belong to the same linter or is the check different enough that it warrants a new linter?
